Code/src/main/java/nl/utwente/viskell/haskell/env/CatalogFunction.java

```diff
@@ -33,6 +37,31 @@
         this.isCommon = isCommon;
     }
 
+    @Override
+    public Map<String, Object> toBundleFragment() {
+        Map<String, Object> bundleFragment = new HashMap<>();
+        bundleFragment.put(Bundleable.KIND, this.getClass().getSimpleName());
+        bundleFragment.put("name", name);
+        bundleFragment.put("signature", signature.toString());
+        bundleFragment.put("category", category);
+        bundleFragment.put("documentation", documentation);
+        bundleFragment.put("isConstructor", isConstructor);
+        bundleFragment.put("isCommon", isCommon);
```

Comment:
Hmm. It could be that some components reference CatalogFunctions where a FunctionInfo would also suffice, which would make this a bit easier. I don't have the tools right now to check that, though.